src/pages/Studio/URLExtension.js

```diff
@@ -98,14 +102,22 @@ const URLExtension = ({
             shortUrlHash = newShortUrlHash
             setShortUrlHash(newShortUrlHash)
 
-            history.push(buildChartShortPath(shortUrlHash))
+            history.push(
+              isNew
+                ? buildChartShortPath(shortUrlHash) + '/new'
+                : buildChartShortPath(shortUrlHash)
+            )
           })
 
       shortUrlPromise
         .then(() => {
           if (isRacing) return
 
-          history.replace(buildChartShortPath(shortUrlHash))
+          history.replace(
+            isNew
+              ? buildChartShortPath(shortUrlHash) + '/new'
+              : buildChartShortPath(shortUrlHash)
```

Comment:
To reduce amount of repeated code we can store the result of `buildChartShortPath` in the variable and conditionally append it, or as a one liner: `buildChartShortPath(shortUrlHash) + (isNew ? '/new' : '')`